Time Domain Analysis Functions.R

```diff
@@ -0,0 +1,433 @@
+
+rm(list=ls())
+
+dataset <- `2019.7.10_RR_p36t0`
+
+data <- as.numeric(dataset[,2])
+x <- data.frame(data)
+
+# Program uses RR rather than NN values
+# RR includes all heart beats while NN is for only "normal" heart beats
+
+# RMSSD FUNCTION
+# most common measure of heart rate variability 
+
+RMSSD_Calculator  <- function(x, input, segment, max, min) {  
+
+  if (input == "RR" && segment == FALSE) {
+    
+    input_length <- length(data)
+    
+    input_length <- input_length - 1
+    
+    RR_squared <- x^2
+    
+    RR_squared_average <- sum(RR_squared) / input_length
+    
+    rmssd1 <- sqrt(RR_squared_average)
+    
+    return(rmssd1)
+    
+  } else if (input == "MS" && segment == FALSE) {
+    
+    RR <- diff(x)
+    
+    RR <- as.numeric(RR)
+    
+    input_length <- length(data)
+    
+    input_length <- input_length - 1
+    
+    RR_squared <- x^2
+    
+    RR_squared_average <- sum(RR_squared) / input_length
+    
+    rmssd1 <- sqrt(RR_squared_average)
+    
+    return(rmssd1)
+    
+  } else if (input == "RR" && segment == TRUE) {
+    
+    input_length <- length(data)
+    
+    input_length <- input_length - 1
+    
+    # create a new vector that converts RR values to time 
+   
+    time_vector <- vector()
+    
+    for (value in 1:nrow(x)) {
+    
+        newsum <- sum(x[1:value, ])
+      
+        time_vector <- append(sumVector, newsum)
+    }
+
+    # millisecond to minutes
+    
+    df1 <- time_vector / 1000
+    
+    # then add this to existing dataframe, so that the user can index time
+    
+    df1_plus_time <- cbind(df1, x)
+    
+    time_constrained_RRvalues <- df1_plus_time[min:max, 2]
+    
+    # back to regular rmssd
+    
+    RR_squared <- time_constrained_RRvalues ^ 2
+    
+    RR_squared_average <- sum(RR_squared) / input_length
+    
+    rmssd1 <- sqrt(RR_squared_average)
+    
+    return(rmssd1)
+    
+  } else if (input == "MS" && segment == TRUE) {
+    
+    RR <- diff(x)
+  
+    RR <- as.numeric(RR)
+  
+    input_length <- length(data)
+  
+    input_length <- input_length - 1
+  
+    # convert MS values to seconds
+  
+    df1 <- x / 1000
+    
+    newdata <- cbind(df1, x)
+    
+    time_constrained_RRvalues <- newdata[min:max, 2]
+    
+    #back to regular rmssd
+    
+    RR_squared <- time_constrained_RRvalues ^ 2
+    
+    RR_squared_average <- sum(RR_squared) / input_length
+    
+    rmssd1 <- sqrt(RR_squared_average)
+    
+    return(rmssd1)
+  
+    }
+
+  }
+
+
+# Plotting RR values data for a general overview 
+
+RR_plot_function <- function (x = data, input =, segment = , min = , max =) {
+  
+  if (input = "RR" && segment = FALSE) {
+  
+    RR <- data[ , 1]  
+    
+    plot(RR, type = "l", xlab = "Seconds", ylab = "RR Values")
+  
+    }
+  
+  else if (input = "MS" && segment = FALSE) {
+    
+    RR <- diff(data [ , 1]) 
+    
+    plot(RR, type = "l", xlab = "Seconds", ylab = "RR Values")
+    
+  }
+  
+  else if (input = "RR" && segment = TRUE) {
```

Comment:
For the next two if - statements, you could create a short function that takes an input (RR or MS) and then plots it accordingly. This will be much shorter than these two if - statements that are basically the same except one line (RR <- data[] / RR <- diff(data[]).